### PR TITLE
Adding Dashboard Data Generator for Case Progress Dashboard

### PR DESCRIPTION
**Pull request by askeing**
draft site: https://askeing.github.io/Hasal_dashboard/

### CODE REVIEW

**Comment by askeing** [comment]
draft site: https://askeing.github.io/Hasal_dashboard/

**Comment by askeing** [comment]
draft site: https://askeing.github.io/Hasal_dashboard/

**Comment by askeing** [comment]
@ShakoHo @ypwalter r?

File: ejenti/server_jobs/dashboard_data_generator.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+                    casename, browser, platform = key.split(':')
+
+                    self.casename_set.add(casename)
+
+                    value_list = value.get(DashboardDataGenerator.KEY_VLAUE)
+                    signature = value.get(DashboardDataGenerator.KEY_SIGNATURE)
+
+                    ret_obj[DashboardDataGenerator.KEY_TIMESTAMP] = timestamp
+                    ret_obj[DashboardDataGenerator.KEY_TIMESTAMP_JS] = int(timestamp) * 1000
+                    ret_obj[DashboardDataGenerator.KEY_CASENAME] = casename
```
Comment (askeing) [inline ejenti/server_jobs/dashboard_data_generator.py]: When there is missing result at latest build, the overall progress will be incorrect.
Should adding a set for storing all `case name` for reference at `generate_latest_build_overall_progress_for_platform()`.

**Comment by askeing** [comment]
@ShakoHo @ypwalter r?

**Comment by askeing** [comment]
@ShakoHo @ypwalter r?

**Review by ypwalter — APPROVED** [review]
r+

**Comment by askeing** [comment]
Dashboard also be merged: https://mozilla-twqa.github.io/Hasal_dashboard/

**Pull request merged**